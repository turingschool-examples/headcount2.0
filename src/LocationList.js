import React from 'react';
import Search from './Search';

const LocationList = ({ displayedLocations, 
                        selectLocation, 
                        searchLocations }) => {
  const locationButtons = displayedLocations.map((location, index) => (
    <button key={index}
            name={location}
            className='LocationList__btn'
            onClick={() => selectLocation(location)} >
      {location}
    </button>
  ));

  return (
    <aside className='LocationList'>
      <Search searchLocations={searchLocations} />
      {locationButtons}
    </aside>
  );
};

export default LocationList;