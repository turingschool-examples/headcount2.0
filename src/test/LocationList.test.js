import React from 'react';
import LocationList from '../LocationList';
import { shallow } from 'enzyme';

describe('LocationList', () => {

    it('should match the snapshot', () => {
        const wrapper = shallow(<LocationList displayedLocations={[]} />)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match the snapshot with one location', () => {
        const wrapper = shallow(<LocationList displayedLocations={['COLORADO']} />)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should match the snapshot with multiple locations', () => {
        const wrapper = shallow(<LocationList displayedLocations={['COLORADO', 'ACADEMY 20', 'AGATE 300']} />)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should call selectLocation when a location button is clicked', () => {
        const selectLocationMock = jest.fn()
        const wrapper = shallow(<LocationList displayedLocations={['COLORADO']} selectLocation={selectLocationMock} />)

        wrapper.find('.LocationList__btn').simulate('click')

        expect(selectLocationMock).toHaveBeenCalled()
    })
})