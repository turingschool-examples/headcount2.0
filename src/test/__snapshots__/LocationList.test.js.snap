// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationList should match the snapshot 1`] = `"<aside class=\\"LocationList\\"><input class=\\"Search\\" placeholder=\\"SEARCH\\"/></aside>"`;

exports[`LocationList should match the snapshot with multiple locations 1`] = `"<aside class=\\"LocationList\\"><input class=\\"Search\\" placeholder=\\"SEARCH\\"/><button name=\\"COLORADO\\" class=\\"LocationList__btn\\">COLORADO</button><button name=\\"ACADEMY 20\\" class=\\"LocationList__btn\\">ACADEMY 20</button><button name=\\"AGATE 300\\" class=\\"LocationList__btn\\">AGATE 300</button></aside>"`;

exports[`LocationList should match the snapshot with one location 1`] = `"<aside class=\\"LocationList\\"><input class=\\"Search\\" placeholder=\\"SEARCH\\"/><button name=\\"COLORADO\\" class=\\"LocationList__btn\\">COLORADO</button></aside>"`;
